polling_stations/apps/api/postcode.py

```diff
@@ -23,32 +24,26 @@ class PostcodeViewSet(ViewSet, LogLookUpMixin):
     lookup_field = "postcode"
     serializer_class = PostcodeResponseSerializer
 
-    def get_object(self, **kwargs):
-        assert "location" in kwargs
-        assert "council" in kwargs
-        return PollingStation.objects.get_polling_station(
-            kwargs["council"], location=kwargs["location"]
-        )
-
     def generate_addresses(self, routing_helper):
         if routing_helper.route_type == "multiple_addresses":
             sorter = AddressSorter(routing_helper.addresses)
             return sorter.natural_sort()
         return []
 
-    def generate_polling_station(self, routing_helper, council, location):
+    def generate_polling_station(self, routing_helper):
         if routing_helper.route_type == "single_address":
             return PollingStation.objects.get_polling_station_by_id(
                 routing_helper.addresses[0].polling_station_id,
-                council_id=routing_helper.addresses[0].council_id,
+                council_id=routing_helper.addresses[0].council,
```

Comment:
You should be able to replace this with just `return routing_helper.addresses[0].polling_station` here and then you can completely bin `PollingStationManager` too. We only need the "get the polling station for an address" logic in one place :)